### PR TITLE
Fix owner kind of stack install job

### LINKED ISSUE #895: GKE: Stacks and CR of stack got deleted for no reason
### What happened?
On a GKE cluster (v1.15.3-gke.1), I installed crossplane (v0.3.0) and deployed wordpress by following the stacks guide (for GCP). Everything was fine and I was able to reach wordpress app from browser. Then, I experimented a bit, several `get's` on different objects, which I expect to be unrelated.

After about 21h, I checked the cluster again but my wordpressinstance seems to be gone. Digging further, I realized that, the stacks (gcp and wordpress) already gone.

```
$ kubectl get stacks --all-namespaces
No resources found
$ kubectl get stackinstalls --all-namespaces
NAMESPACE          NAME                     READY   SOURCE   PACKAGE                                    CRD   AGE
app-project1-dev   sample-stack-wordpress   True             crossplane/sample-stack-wordpress:latest         20h
$ kubectl get clusterstackinstalls --all-namespaces
NAMESPACE   NAME        READY   SOURCE   PACKAGE                       CRD   AGE
gcp         stack-gcp   True             crossplane/stack-gcp:master         21h
```

Most of the CRDs belonging to stacks also gone but not all:

```
$ kubectl get crds | grep gcp
cloudsqlinstances.database.gcp.crossplane.io            2019-10-06T10:41:37Z
subnetworks.compute.gcp.crossplane.io                   2019-10-06T10:41:37Z
```

When I check the logs of stack-manager, I see some connectivity issues at some point in time but finally recovered which could be related to the issue.

```
2019-10-07T02:40:47.548Z	DEBUG	controller-runtime.controller	Successfully Reconciled	{"controller": "stackinstall.stacks.crossplane.io", "request": "app-project1-dev/sample-stack-wordpress"}
2019-10-07T02:40:47.553Z	DEBUG	crossplane.stackinstall.stacks.crossplane.io	reconciling	{"kind": "stacks.crossplane.io/v1alpha1, Kind=ClusterStackInstall", "request": "gcp/stack-gcp"}
2019-10-07T02:40:47.553Z	DEBUG	crossplane.util	getting pod	{"name": "crossplane-system/crossplane-stack-manager-7fd8d6b9fc-mspt7"}
2019-10-07T02:40:47.553Z	DEBUG	crossplane.stackinstall.stacks.crossplane.io	updating not supported yet	{"clusterstackinstall.stacks.crossplane.io/v1alpha1": "gcp/stack-gcp"}
2019-10-07T02:40:47.553Z	DEBUG	controller-runtime.controller	Successfully Reconciled	{"controller": "clusterstackinstall.stacks.crossplane.io", "request": "gcp/stack-gcp"}
E1007 02:49:27.141062       1 reflector.go:270] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:126: Failed to watch *v1alpha1.Stack: Get https://10.8.0.1:443/apis/stacks.crossplane.io/v1alpha1/stacks?resourceVersion=125697&timeoutSeconds=364&watch=true: dial tcp 10.8.0.1:443: connect: connection refused
E1007 02:49:27.142221       1 reflector.go:270] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:204: Failed to watch *v1.Pod: Get https://10.8.0.1:443/api/v1/pods?resourceVersion=180641&timeoutSeconds=386&watch=true: dial tcp 10.8.0.1:443: connect: connection refused
E1007 02:49:27.142289       1 reflector.go:270] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:126: Failed to watch *v1alpha1.ClusterStackInstall: Get https://10.8.0.1:443/apis/stacks.crossplane.io/v1alpha1/clusterstackinstalls?resourceVersion=1582&timeoutSeconds=423&watch=true: dial tcp 10.8.0.1:443: connect: connection refused
E1007 02:49:27.142340       1 reflector.go:270] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:126: Failed to watch *v1alpha1.StackInstall: Get https://10.8.0.1:443/apis/stacks.crossplane.io/v1alpha1/stackinstalls?resourceVersion=3464&timeoutSeconds=450&watch=true: dial tcp 10.8.0.1:443: connect: connection refused
E1007 02:49:27.142393       1 reflector.go:270] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:204: Failed to watch *v1.Job: Get https://10.8.0.1:443/apis/batch/v1/jobs?resourceVersion=125695&timeoutSeconds=397&watch=true: dial tcp 10.8.0.1:443: connect: connection refused
E1007 02:49:28.143938       1 reflector.go:126] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:126: Failed to list *v1alpha1.Stack: Get https://10.8.0.1:443/apis/stacks.crossplane.io/v1alpha1/stacks?limit=500&resourceVersion=0: dial tcp 10.8.0.1:443: connect: connection refused
E1007 02:49:28.145037       1 reflector.go:126] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:204: Failed to list *v1.Pod: Get https://10.8.0.1:443/api/v1/pods?limit=500&resourceVersion=0: dial tcp 10.8.0.1:443: connect: connection refused
E1007 02:49:28.146116       1 reflector.go:126] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:126: Failed to list *v1alpha1.ClusterStackInstall: Get https://10.8.0.1:443/apis/stacks.crossplane.io/v1alpha1/clusterstackinstalls?limit=500&resourceVersion=0: dial tcp 10.8.0.1:443: connect: connection refused
E1007 02:49:28.147583       1 reflector.go:126] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:126: Failed to list *v1alpha1.StackInstall: Get https://10.8.0.1:443/apis/stacks.crossplane.io/v1alpha1/stackinstalls?limit=500&resourceVersion=0: dial tcp 10.8.0.1:443: connect: connection refused
...
E1007 02:50:01.220220       1 reflector.go:126] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:126: Failed to list *v1alpha1.StackInstall: Get https://10.8.0.1:443/apis/stacks.crossplane.io/v1alpha1/stackinstalls?limit=500&resourceVersion=0: dial tcp 10.8.0.1:443: connect: connection refused
E1007 02:50:02.217546       1 reflector.go:126] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:126: Failed to list *v1alpha1.ClusterStackInstall: Get https://10.8.0.1:443/apis/stacks.crossplane.io/v1alpha1/clusterstackinstalls?limit=500&resourceVersion=0: dial tcp 10.8.0.1:443: connect: connection refused
E1007 02:50:02.218231       1 reflector.go:126] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:126: Failed to list *v1alpha1.Stack: Get https://10.8.0.1:443/apis/stacks.crossplane.io/v1alpha1/stacks?limit=500&resourceVersion=0: dial tcp 10.8.0.1:443: connect: connection refused
E1007 02:50:02.219803       1 reflector.go:126] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:204: Failed to list *v1.Job: Get https://10.8.0.1:443/apis/batch/v1/jobs?limit=500&resourceVersion=0: dial tcp 10.8.0.1:443: connect: connection refused
E1007 02:50:02.221845       1 reflector.go:126] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:126: Failed to list *v1alpha1.StackInstall: Get https://10.8.0.1:443/apis/stacks.crossplane.io/v1alpha1/stackinstalls?limit=500&resourceVersion=0: dial tcp 10.8.0.1:443: connect: connection refused
E1007 02:50:02.221959       1 reflector.go:126] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:204: Failed to list *v1.Pod: Get https://10.8.0.1:443/api/v1/pods?limit=500&resourceVersion=0: dial tcp 10.8.0.1:443: connect: connection refused
2019-10-07T02:50:12.659Z	DEBUG	crossplane.stackinstall.stacks.crossplane.io	reconciling	{"kind": "stacks.crossplane.io/v1alpha1, Kind=ClusterStackInstall", "request": "gcp/stack-gcp"}
2019-10-07T02:50:12.659Z	DEBUG	crossplane.util	getting pod	{"name": "crossplane-system/crossplane-stack-manager-7fd8d6b9fc-mspt7"}
2019-10-07T02:50:12.659Z	DEBUG	crossplane.stackinstall.stacks.crossplane.io	updating not supported yet	{"clusterstackinstall.stacks.crossplane.io/v1alpha1": "gcp/stack-gcp"}
2019-10-07T02:50:12.659Z	DEBUG	controller-runtime.controller	Successfully Reconciled	{"controller": "clusterstackinstall.stacks.crossplane.io", "request": "gcp/stack-gcp"}
2019-10-07T02:50:12.909Z	DEBUG	crossplane.stackinstall.stacks.crossplane.io	reconciling	{"kind": "stacks.crossplane.io/v1alpha1, Kind=StackInstall", "request": "app-project1-dev/sample-stack-wordpress"}
2019-10-07T02:50:12.909Z	DEBUG	crossplane.util	getting pod	{"name": "crossplane-system/crossplane-stack-manager-7fd8d6b9fc-mspt7"}
2019-10-07T02:50:12.909Z	DEBUG	crossplane.stackinstall.stacks.crossplane.io	updating not supported yet	{"stackinstall.stacks.crossplane.io/v1alpha1": "app-project1-dev/sample-stack-wordpress"}
2019-10-07T02:50:12.909Z	DEBUG	controller-runtime.controller	Successfully Reconciled	{"controller": "stackinstall.stacks.crossplane.io", "request": "app-project1-dev/sample-stack-wordpress"}
2019-10-07T03:50:12.659Z	DEBUG	crossplane.stackinstall.stacks.crossplane.io	reconciling	{"kind": "stacks.crossplane.io/v1alpha1, Kind=ClusterStackInstall", "request": "gcp/stack-gcp"}
2019-10-07T03:50:12.659Z	DEBUG	crossplane.util	getting pod	{"name": "crossplane-system/crossplane-stack-manager-7fd8d6b9fc-mspt7"}
2019-10-07T03:50:12.659Z	DEBUG	crossplane.stackinstall.stacks.crossplane.io	updating not supported yet	{"clusterstackinstall.stacks.crossplane.io/v1alpha1": "gcp/stack-gcp"}
2019-10-07T03:50:12.659Z	DEBUG	controller-runtime.controller	Successfully Reconciled	{"controller": "clusterstackinstall.stacks.crossplane.io", "request": "gcp/stack-gcp"}
2019-10-07T03:50:12.909Z	DEBUG	crossplane.stackinstall.stacks.crossplane.io	reconciling	{"kind": "stacks.crossplane.io/v1alpha1, Kind=StackInstall", "request": "app-project1-dev/sample-stack-wordpress"}
```

Some more info, which could help:
```
$ kubectl get mysqlinstances --all-namespaces
NAMESPACE          NAME                                                   STATUS   CLASS            VERSION   AGE
app-project1-dev   wordpress-mysql-09210911-2260-4106-a01c-9b860aefa24b   Bound    standard-mysql   5.7       21h
$ kubectl get cloudsqlinstances --all-namespaces
NAMESPACE   NAME                                                 STATUS   STATE   CLASS               VERSION     AGE
gcp         mysqlinstance-e55cd378-4009-4274-92ee-43b4f268154e                    standard-cloudsql   MYSQL_5_7   11h
$ kubectl get kubernetesclusters --all-namespaces
NAMESPACE          NAME                                                     STATUS   CLUSTER-CLASS      CLUSTER-REF   AGE
app-project1-dev   wordpress-cluster-09210911-2260-4106-a01c-9b860aefa24b   Bound    standard-cluster                 21h
$ kubectl get gkecluster --all-namespaces
error: the server doesn't have a resource type "gkecluster"
$ kubectl get kubernetesapplications --all-namespaces
No resources found
$ kubectl get kubernetesapplicationresources --all-namespaces
NAMESPACE          NAME                                                             TEMPLATE-KIND   TEMPLATE-NAME   CLUSTER                                                  STATUS
app-project1-dev   wordpress-demo-deployment-09210911-2260-4106-a01c-9b860aefa24b   Deployment      wordpress       wordpress-cluster-09210911-2260-4106-a01c-9b860aefa24b   Submitted
app-project1-dev   wordpress-demo-namespace-09210911-2260-4106-a01c-9b860aefa24b    Namespace       wordpress       wordpress-cluster-09210911-2260-4106-a01c-9b860aefa24b   Submitted
app-project1-dev   wordpress-demo-service-09210911-2260-4106-a01c-9b860aefa24b      Service         wordpress       wordpress-cluster-09210911-2260-4106-a01c-9b860aefa24b   Submitted
$ kubectl get wordpressinstances --all-namespaces
error: the server doesn't have a resource type "wordpressinstances"
```
Restarting stack manager pod didn't help.

### How can we reproduce it?

Follow stacks guide, leave the cluster. If the issue is related to aforementioned connectivity issue, breaking connectivity betweek stack manager and api server could help to reproduce.

### What environment did it happen in?
Crossplane version: v0.3.0

### PR DESCRIPTION
**Pull request by turkenh**
Signed-off-by: Hasan Turken <turkenh@gmail.com>

### Description of your changes

There is a wrong object reference in stack install jobs. It contains the uid of ClusterStackInstall object but kind/name of Stack object. This commit changes kind as ClusterStackInstall.

Fixes #895

### Checklist

I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [ ] Updated any relevant [documentation], [examples], or [release notes].
- [ ] Updated the RBAC permissions in [`clusterrole.yaml`] to include any new types.

[documentation]: https://github.com/crossplaneio/crossplane/tree/master/docs
[examples]: https://github.com/crossplaneio/crossplane/tree/master/cluster/examples
[release notes]: https://github.com/crossplaneio/crossplane/tree/master/PendingReleaseNotes.md
[`clusterrole.yaml`]: https://github.com/crossplaneio/crossplane/blob/master/cluster/charts/crossplane/templates/clusterrole.yaml

### CODE REVIEW

File: pkg/controller/stacks/install/installjob.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ type stackInstallJobCompleter struct {
 }
 
 func createInstallJob(i v1alpha1.StackInstaller, executorInfo *stacks.ExecutorInfo) *batchv1.Job {
-	ref := meta.AsOwner(meta.ReferenceTo(i, v1alpha1.StackGroupVersionKind))
+	ref := meta.AsOwner(meta.ReferenceTo(i, i.GroupVersionKind()))
```
Comment (jbw976) [inline pkg/controller/stacks/install/installjob.go]: This fix makes sense.  do we have the same issue on these lines too?

* https://github.com/crossplaneio/crossplane/blob/master/pkg/controller/stacks/stack/stack.go#L229
* https://github.com/crossplaneio/crossplane/blob/master/pkg/controller/stacks/stack/stack.go#L264
* https://github.com/crossplaneio/crossplane/blob/master/pkg/controller/stacks/stack/stack.go#L299

Comment (displague) [reply pkg/controller/stacks/install/installjob.go]: I think the other locations make sense.

When the Stack Manager creates the Deployment or Job or RBAC defined in the Stack fields it must set the owner of those resources to the Stack itself (`h.ext`) and use the GVK of a "Stack" -- `v1alpha1.StackGroupVersionKind`.

`createInstallJob` is run earlier in the processing of a Stack. The StackInstall manager intends to create the `Job` that will emits the Stack and CRD yaml resources.  This `Job` should reference the `StackInstall` or `ClusterStackInstall` resource that it is related to, and here the object and type are mismatched.

Nice find, @turkenh!

Comment (turkenh) [reply pkg/controller/stacks/install/installjob.go]: Agree with @displague. Other lines look good to me.

Comment (jbw976) [reply pkg/controller/stacks/install/installjob.go]: ah, got it, thank you for confirming those other instances @displague and @turkenh!

**Pull request merged**